File: solutions/leetcode/move-zeroes.cpp
/* Problem move-zeroes
 * https://www.leetcode.com/problems/move-zeroes
 */

#include <algorithm>
#include <vector>

using namespace std;

class Solution {
 public:
  void moveZeroes(vector<int>& nums) {
    fill(remove(nums.begin(), nums.end(), 0), nums.end(), 0);
  }
};

File: solutions/leetcode/move-zeroes.rs
/* Problem move-zeroes
 * https://www.leetcode.com/problems/move-zeroes
 */

struct Solution;

impl Solution {
	pub fn move_zeroes(nums: &mut Vec<i32>) {
		let mut nz = 0;
		for i in 0..nums.len() {
			if nums[i] != 0 {
				nums.swap(i, nz);
				nz += 1;
			}
		}
	}
}

fn main() {}
